src/org/labkey/targetedms/chromlib/LibPeptideStructuralModification.java

```diff
@@ -98,8 +98,8 @@ public boolean equals(Object o)
     @Override
     public int hashCode()
     {
-        int result = _peptideId;
-        result = 31 * result + _structuralModificationId;
+        int result = Math.toIntExact(_peptideId);
```

Comment:
This will blow up with a ArithmeticException if the ID value exceeds MAX_INT. Please double-check other new uses of Math.toIntExact as well.

For hash code purposes, I think we can just simply cast to an int - it's fine if it overflows.